tensorflow/compiler/tf2xla/kernels/reduction_ops.h

```diff
@@ -39,6 +39,10 @@ class XlaReductionOp : public XlaOpKernel {
   // Return the base case for the reduction.
   virtual xla::XlaOp InitialValue(xla::XlaBuilder* builder) = 0;
 
+  // Processes data before reduction.
+  virtual xla::XlaOp ProcessData(xla::XlaBuilder* builder,
+                                 const xla::XlaOp& data);
+
```

Comment:
Should be called `PreprocessInput` or something like that.  `ProcessData` sounds super generic.